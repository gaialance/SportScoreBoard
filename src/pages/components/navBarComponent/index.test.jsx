// import dependencies
import { render,screen , cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import '@testing-library/jest-dom/extend-expect'
import NavBarComponent from '.'
import { DateTime } from 'luxon'

afterEach(cleanup);

describe("rendering the NavBarComponent",()=>{
      it('making Snapsnot the landing page to compare',() => {
         const { container } = render(<NavBarComponent />)

         expect(container).toMatchSnapshot()
      });

     it('testing the dateformat in the screen correct or not at the start',() => {
         render( <NavBarComponent/> )

        expect(  screen.getByTestId("dateFormat")  ).toHaveTextContent("Today")
     });

     it('getting the date filter that is before that day when click on the previous button it will minus one day', async () => {
         render( <NavBarComponent/> )

        expect( screen.getByTestId("dateFormat")  ).toHaveTextContent("Today")
    
        await userEvent.click(screen.getByTestId('prevButton'))

        expect( screen.getByTestId("dateFormat") ).toHaveTextContent(DateTime.now().minus({day:1}).toLocaleString().toString())

     });

     it('getting the date filter that is before that day when click on the previous button it will add one day', async () => {
         render( <NavBarComponent/> )  

        expect( screen.getByTestId("dateFormat")  ).toHaveTextContent("Today")
    
        await userEvent.click(screen.getByTestId('nextButton'))

        expect( screen.getByTestId("dateFormat") ).toHaveTextContent(DateTime.now().plus({day:1}).toLocaleString().toString())

     });

})