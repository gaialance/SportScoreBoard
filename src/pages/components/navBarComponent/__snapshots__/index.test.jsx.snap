// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering the NavBarComponent making Snapsnot the landing page to compare 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      class="container MuiBox-root css-0"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 navigationButton css-1o7apob-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium button css-l0ycqr-MuiButtonBase-root-MuiButton-root"
            data-testid="prevButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 textMiddle css-vok43g-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium button css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              data-testid="dateFormat"
            >
              Today
            </h5>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropUpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m7 14 5-5 5 5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 navigationButton css-1o7apob-MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium button css-l0ycqr-MuiButtonBase-root-MuiButton-root"
            data-testid="nextButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
